tree/treeplayer/src/TDFInterfaceUtils.cxx

```diff
@@ -155,20 +155,11 @@ SelectColumns(unsigned int nRequiredNames, const ColumnNames_t &names, const Col
 
 bool IsTreeLeaf(TTree &t, const std::string &leaf)
 {
-   // TODO understand why GetBranch is also needed (run the tests without, inspect failures)
```

Comment:
Note searching on the branch means that you might no longer be given access to the object if it is split (i.e. the event itself for example).

Reply:
sorry I don't understand what you mean, could you elaborate a bit more?

Reply:
If you only look in the list of leaves and do not do any 'massaging'.   For the example in https://sft.its.cern.ch/jira/browse/ROOT-9312, (after adding member functions) you would not be able to detect "w.GetCalcucatedValue()" as a valid user request since the list of leaves only contains the end nodes.